src/charm.py

```diff
@@ -509,12 +524,25 @@ def _get_primary_address_from_peers(self) -> str:
                 return self.peers.data[unit]["instance-hostname"]
 
     def _restart(self, _) -> None:
+        """Restart server rolling ops callback function.
+
+        Hold execution until server is back in the cluster.
+        Used exclusively for rolling restarts.
+        """
+        logger.debug("Restarting mysqld daemon")
         if service_restart(SERVICE_NAME):
```

Comment:
what should we do if the service restart is not successful?

Reply:
Well spotted. Definitely block the unit ( 7a72c73 )